packages/subgraph/schema.graphql

```diff
@@ -236,3 +237,46 @@ type AllowlistProposal implements Proposal @entity {
   createdAt: BigInt!
   executable: Boolean
 }
+
+type PluginSetup @entity(immutable: true) {
+  id: ID!
+  versions: [PluginVersion!]! @derivedFrom(field: "pluginSetup")
+  installations: [Plugin!]! @derivedFrom(field: "pluginSetup")
+}
+
+type PluginVersion @entity(immutable: true) {
+  id: ID! # pluginRepo + version index
+  pluginRepo: PluginRepo!
+  semanticVersion: [BigInt!]!
+  pluginSetup: PluginSetup!
+  contentURI: Bytes!
+}
+
+type PluginRepo @entity(immutable: true) {
+  id: ID! # address
+  name: String!
+  versions: [PluginVersion!]! @derivedFrom(field: "pluginRepo")
+}
+
+type PluginHelper @entity(immutable: true) {
+  id: ID! # address
+  plugin: Plugin!
+}
+
+type Plugin @entity {
+  id: ID! # address
+  dao: Dao!
+  pluginSetup: PluginSetup!
+  data: Bytes!
+  helpers: [PluginHelper!]! @derivedFrom(field: "plugin")
+  state: PluginState!
+  sender: String!
+}
+
+enum PluginState {
+  InstallationPending
```

Comment:
Is `XYZPending` a good name? In `PluginSetupProcessor` we call it `XYZPrepared`

Reply:
good point. done